subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -1973,17 +1961,16 @@ static int handle_request(struct zoap_packet *request,
 		format = zoap_option_value_to_int(&options[0]);
 	} else {
 		SYS_LOG_DBG("No content-format given. Assume text plain.");
-		format = LWM2M_FORMAT_OMA_PLAIN_TEXT;
+		format = LWM2M_FORMAT_PLAIN_TEXT;
 	}
 
 	/* read Accept */
 	r = zoap_find_options(in.in_zpkt, ZOAP_OPTION_ACCEPT, options, 1);
 	if (r > 0) {
 		accept = zoap_option_value_to_int(&options[0]);
 	} else {
-		SYS_LOG_DBG("No Accept header: use same as content-format(%d)",
-			    format);
-		accept = format;
+		SYS_LOG_DBG("No accept option given. Assume OMA TLV.");
+		accept = LWM2M_FORMAT_OMA_TLV;
```

Comment:
Update the commit description to explain the modification of choosing OMV TLV as a default accept format.